### DISCUSSION TITLE
Unstable Rate shouldn't be converted in osu!mania

### DISCUSSION THREAD

**Discussion by KapiWilq** [ppy/osu#30698] [Q&A]
A year ago, [there was introduced a change in the way that the Unstable Rate is calculated](https://github.com/ppy/osu/pull/25415). It made the Unstable Rate account for the gameplay rate, which made the Unstable Rate not require conversion when playing with rate-changing mods. [There was raised a fair concern about osu!mania](https://github.com/ppy/osu/pull/25415#issuecomment-1807452912), but according to [these](https://github.com/ppy/osu/pull/25415#issuecomment-1811435714) [replies](https://github.com/ppy/osu/pull/25415#issuecomment-1815833239) the Unstable Rate value for osu!mania should be correct. It's not correct.

## Why?
osu!mania scales hit windows to account for the rate change[^1][^2]. This makes the Unstable Rate not require conversion when playing with rate-changing mods since hit windows are being resized to make them "effectively" the same size as if you are playing without rate-changing mods. Therefore, by converting the Unstable Rate in osu!mania, it's value is incorrect. This also makes the Unstable Rate value shown in osu!(stable) correct.

| osu!(stable) | osu!(lazer) |
| - | - |
| ![screenshot352](https://github.com/user-attachments/assets/219f5f29-a917-4d23-b1d0-f26e2b77aca2) | ![osu_2024-11-16_18-15-38](https://github.com/user-attachments/assets/7be2de05-2a13-40b3-b30c-b3ad9d8b6eb7) |

This is the score that I used for the comparison in these screenshots: https://osu.ppy.sh/scores/mania/301846710

[^1]: https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Mania/Mods/IManiaRateAdjustmentMod.cs#L28
[^2]: https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Mania/Scoring/ManiaHitWindows.cs#L39-L45

**Comment by peppy** [comment]
I made https://github.com/ppy/osu/issues/30828, but it turns out there's not a single objective correct answer here. Can discuss further in this thread.

**Comment by KapiWilq** [comment]
The discussion was initially based on the assumption that the Unstable Rate is related in some way to hit windows, which (after looking at the very definition of Unstable Rate, and therefore the definition of the standard deviation) is not, since you could have the same hit error distribution regardless of hit windows' sizes (even after scaling to account for rate-change). Please note that this is coming from a non-mania player, which is why I got confused about the hit windows being scaled only in osu!mania, and Unstable Rate being converted regardless. After reading the comments in the issue thread I can now see and understand why the conversion of the UR should be in osu!mania after all.

I stand corrected, this discussion can be ended/closed now in my opinion, if no one feels like collecting opinions from osu!mania players.
